src/cloud/components/Views/Table/TableViewContentManager.tsx

```diff
@@ -252,6 +254,67 @@ const TableViewContentManager = ({
     [setPreferences]
   )
 
+  const [newFolderRowState, setNewFolderRowState] = useState<
```

Comment:
Both DocRow and Folder row essentially have the exact same behavior
I would personally extract this into its own component
`TableViewCreationRow` and just have the api calls +. labels forwarded by props.